File: app/models/user_model.py
class UserModel:
    def __init__(self, username, name, role):
        self.username = username
        self.name = name
        self.role = role

    def to_dict(self):
        return {
            "username": self.username,
            "name": self.name,
            "role": self.role,
        }

File: lib/data/models/user_model.dart
class UserModel {
  final String username;
  final String name;
  final String role;

  UserModel({
    required this.username,
    required this.name,
    required this.role,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) => UserModel(
    username: json['username'],
    name: json['name'],
    role: json['role'],
  );

  Map<String, dynamic> toJson() => {
    'username': username,
    'name': name,
    'role': role,
  };
}